### PR TITLE
Game Tournament Scenario

### PR DESCRIPTION
**Pull request by cmeiklejohn**
- [x] Game Tournament Server
- [x] Game Tournament Client
- [x] Game Tournament Supervision
- [x] Game Tournament Tests
- [x] Reduce duplication between scenarios

### CODE REVIEW

File: src/lasp_game_tournament_client.erl [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+                        error ->
+                            Pid = spawn_link(fun() -> trigger(GameId, Actor) end),
+                            dict:store(GameId, Pid, Triggers0)
+                    end;
+                false ->
+                    Triggers0
+            end,
+
+            %% Enroll random player identifier into the game.
+            PlayerId = lasp_support:puniform(1000),
```
Comment (vitorenesduarte) [inline src/lasp_game_tournament_client.erl]: Should we use `lasp_unique:unique()` here?

File: src/lasp_game_tournament_server.erl [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+%%% gen_server callbacks
+%%%===================================================================
+
+%% @private
+-spec init([term()]) -> {ok, #state{}}.
+init([]) ->
+    lager:info("Game tournament server initialized."),
+
+    %% Delay for graph connectedness.
+    wait_for_connectedness(),
```
Comment (vitorenesduarte) [inline src/lasp_game_tournament_server.erl]: I don't know if it makes sense to also move this to `lasp_marathon_simulations`

**Comment by vitorenesduarte** [comment]
:+1:

File: src/lasp_game_tournament_server.erl (hunk 2 above)
Comment (cmeiklejohn) [inline src/lasp_game_tournament_server.erl]: I'll do so in a seperate commit.

File: src/lasp_game_tournament_server.erl [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+    ),
+
+    NodesWithGamesEnrolled = lists:filter(
+        fun({_Node, {GamesEnrolled, _LogsPushed}}) ->
+            GamesEnrolled
+        end,
+        GamesEnrolledAndLogs
+    ),
+
+    lager:info("Checking for simulation end: ~p nodes with enrolledgames and ~p nodes with logs pushed.",
```
Comment (cmeiklejohn) [inline src/lasp_game_tournament_server.erl]: Fixed in the other commit.

**Comment by cmeiklejohn** [comment]
Once this passes, can you start a run like we did last night to see what it looks like?

File: src/lasp_game_tournament_client.erl (hunk 1 above)
Comment (vitorenesduarte) [inline src/lasp_game_tournament_client.erl]: But that means that an invariant violation could happen and not be accounted;

**Pull request merged**